pkg/broker/ingress/handler.go

```diff
@@ -140,10 +135,13 @@ func (h *Handler) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Re
 		if errors.Is(res, ErrNotFound) {
 			statusCode = nethttp.StatusNotFound
 		}
-		response.WriteHeader(statusCode)
-		response.Write([]byte(msg))
+		nethttp.Error(response, msg, statusCode)
 		return
 	}
+
+	// According to the data plane spec (https://github.com/knative/eventing/blob/master/docs/spec/data-plane.md), a
+	// non-callable SINK (which broker is) MUST respond with 202 Accepted if the request is accepted.
+	response.WriteHeader(nethttp.StatusAccepted)
```

Comment:
Is it safe to hardcode StatusAccepted here or should this be set to statusCode?

Reply:
It's safer to return hardcode 202 here since this is the end of the request with no error. Although statusCode should also be 202 here. Let me know if you have other concerns.

Reply:
I refactored it a little bit. Let me know if this is better.